dotnet.hcl

```diff
@@ -0,0 +1,58 @@
+description = ".NET is a free, cross-platform, open-source developer platform for building many different types of applications."
+homepage = "https://dotnet.microsoft.com/"
+binaries = ["dotnet"]
+strip = 1
+test = "dotnet --version"
+
+vars = {
+    "baseurl": "https://download.visualstudio.microsoft.com/download/pr",
+    "os_": "${os}",
+    "arch_": "${xarch}",
+}
+
+platform "darwin" {
+  vars = {
+    "os_": "osx",
+  }
+}
+
+platform "amd64" {
+  vars = {
+    "arch_": "x64",
+  }
+}
+
+platform "arm64" {
+    vars = {
+        "arch_": "arm64"
+    }
+}
+
+// parse from https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/6.0/releases.json etc
```

Comment:
Not ideal, but what I've done for this in the past is include a shell script fragment as a comment that will return the correct fragments. This may or may not be possible.